base/uk.ac.stfc.isis.ibex.epics.tests/src/uk/ac/stfc/isis/ibex/epics/tests/conversion/ExponentialOnThresholdFormatTest.java

```diff
@@ -0,0 +1,375 @@
+ * EXCEPT AS EXPRESSLY SET FORTH IN THE ECLIPSE PUBLIC LICENSE V1.0, THE PROGRAM 
+ * AND ACCOMPANYING MATERIALS ARE PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES 
+ * OR CONDITIONS OF ANY KIND.  See the Eclipse Public License v1.0 for more details.
+ *
+ * You should have received a copy of the Eclipse Public License v1.0
+ * along with this program; if not, you can obtain a copy from
+ * https://www.eclipse.org/org/documents/epl-v10.php or 
+ * http://opensource.org/licenses/eclipse-1.0.php
+ */
+
+/**
+ * 
+ */
+package uk.ac.stfc.isis.ibex.epics.tests.conversion;
+
+import static org.junit.Assert.assertEquals;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+
+import org.junit.Test;
+
+import uk.ac.stfc.isis.ibex.epics.conversion.ExponentialOnThresholdFormat;
+
+@SuppressWarnings({ "checkstyle:methodname", "checkstyle:magicnumber" })
+public class ExponentialOnThresholdFormatTest {
+
+    private static final String THREE_FRACTION_DIGITS_PATTERN = "0.000";
+    private static final String FIVE_FRACTION_DIGITS_PATTERN = "0.00000";
+    private static final String INTEGER_PATTERN = "#.##";
+
+    private NumberFormat threeFractionalDigitsFormat = new DecimalFormat(THREE_FRACTION_DIGITS_PATTERN);
+    private NumberFormat fiveFractionalDigitsFormat = new DecimalFormat(FIVE_FRACTION_DIGITS_PATTERN);
+    private NumberFormat integerFormat = new DecimalFormat(INTEGER_PATTERN);
+
+    private void verifyNonExponentialFormat(double value, String expected, NumberFormat sourceFormat) {
+        // Arrange
+        NumberFormat format = new ExponentialOnThresholdFormat(sourceFormat);
+
+        // Act-Assert
+        assertEquals(expected, format.format(value));
+        assertEquals(sourceFormat.format(value), format.format(value));
+    }
+
+    private void verifyExponentialFormat(double value, String expected, NumberFormat sourceFormat) {
+        // Arrange
+        NumberFormat format = new ExponentialOnThresholdFormat(sourceFormat);
+
+        // Act-Assert
+        assertEquals(expected, format.format(value));
+    }
+
+    private void verifyNonExponentialFormat(int value, String expected, NumberFormat sourceFormat) {
+        // Arrange
+        NumberFormat format = new ExponentialOnThresholdFormat(sourceFormat);
+
+        // Act-Assert
+        assertEquals(expected, format.format(value));
+        assertEquals(sourceFormat.format(value), format.format(value));
+    }
+
+    private void verifyExponentialFormat(int value, String expected, NumberFormat sourceFormat) {
+        // Arrange
+        NumberFormat format = new ExponentialOnThresholdFormat(sourceFormat);
+
+        // Act-Assert
+        assertEquals(expected, format.format(value));
+    }
+
+    @Test
+    public void
+            GIVEN_value_0_1_above_lower_threshold_WHEN_formatting_with_precision_3_THEN_no_exponential_notation_used() {
+        // Arrange
+        double value = 0.1;
```

Comment:
Surely this value is absolute, not relative to the lower threshold as the name suggests. Please change the test or the test name appropriately. This applies to several subsequent tests